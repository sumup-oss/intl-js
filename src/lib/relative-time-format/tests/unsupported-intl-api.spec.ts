/**
 * Copyright 2022, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { vi, describe, it, expect } from 'vitest';

import {
  formatRelativeTime,
  formatRelativeTimeToParts,
  resolveRelativeTimeFormat,
} from '../index.js';

import { relativeTimes } from './shared.js';

vi.mock('../intl', async () => {
  const intl = await vi.importActual('../intl');
  return {
    ...intl,
    isRelativeTimeFormatSupported: false,
  };
});

const locale = 'xx-XX';

describe('Relative times', () => {
  describe('when Intl.RelativeTimeFormat is unsupported', () => {
    it.each(relativeTimes)(
      'should format %s %s as a relative time',
      (value, unit) => {
        const actual = formatRelativeTime(value, unit, locale);
        expect(actual).toMatchSnapshot();
      },
    );

    it('should format a relative time to numeric parts', () => {
      const [value, unit] = relativeTimes[0];
      const parts = formatRelativeTimeToParts(value, unit, locale);
      expect(parts).toHaveLength(10);
    });

    it('should return `null` for the relative time format', () => {
      const actual = resolveRelativeTimeFormat(locale);
      expect(actual).toBeNull();
    });
  });
});
