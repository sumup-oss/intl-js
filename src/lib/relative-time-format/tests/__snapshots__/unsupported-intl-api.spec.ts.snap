// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Relative times > when Intl.RelativeTimeFormat is unsupported > should format -5 months as a relative time 1`] = `"-5 months"`;

exports[`Relative times > when Intl.RelativeTimeFormat is unsupported > should format 1 day as a relative time 1`] = `"1 day"`;

exports[`Relative times > when Intl.RelativeTimeFormat is unsupported > should format 3 quarters as a relative time 1`] = `"9 months"`;

exports[`Relative times > when Intl.RelativeTimeFormat is unsupported > should format 7 years as a relative time 1`] = `"7 years"`;
